pkg/gpt/gpt.go

```diff
@@ -74,20 +74,52 @@ func (g *GPT) String() string {
 	return string(b)
 }
 
+func errAppend(err error, s string, a ...interface{}) error {
+	var p string
+	if err != nil {
+		p = err.Error() + "; "
+	}
+	return fmt.Errorf(p+s, a...)
+}
+
 // EqualHeader compares two headers and returns true if they match.
 // Those fields which by definition must differ are ignored.
-func EqualHeader(p, b Header) bool {
-	return p.Signature == b.Signature &&
-		p.Revision == b.Revision &&
-		p.HeaderSize == b.HeaderSize &&
-		p.CurrentLBA == b.BackupLBA &&
-		p.BackupLBA == b.CurrentLBA &&
-		p.FirstLBA == b.FirstLBA &&
-		p.LastLBA == b.LastLBA &&
-		p.DiskGUID == b.DiskGUID &&
-		p.NPart == b.NPart &&
-		p.PartSize == b.PartSize &&
-		p.PartCRC == b.PartCRC
+func EqualHeader(p, b Header) error {
+	var err error
+	if p.Signature != b.Signature {
+		err = errAppend(err, "p.Signature(%#x) != b.Signature(%#x)", p.Signature, b.Signature)
+	}
+	if p.Revision != b.Revision {
+		err = errAppend(err, "p.Revision(%v) != b.Revision(%v)", p.Revision, b.Revision)
+	}
+	if p.HeaderSize != b.HeaderSize {
+		err = errAppend(err, "p.HeaderSize(%v) != b.HeaderSize(%v)", p.HeaderSize, b.HeaderSize)
+	}
+	if p.CurrentLBA != b.BackupLBA {
+		err = errAppend(err, "p.CurrentLBA(%#x) != b.BackupLBA(%#x)", p.CurrentLBA, b.BackupLBA)
+	}
+	if p.BackupLBA != b.CurrentLBA {
```

Comment:
same?

Reply:
Nope, same rule. Backup for P is Current for b, and vice-versa.

Reply:
Maybe it should be called something other than `Equal` then? `CorrespondingHeaders`? (too long) Idk. If you don't have better ideas, it looks good to merge.